src/grisp_tools_build.erl

```diff
@@ -63,24 +63,23 @@ build(S0) ->
 
 download(#{otp_version := {_,_,_,Ver}} = S0) ->
     BuildPath = mapz:deep_get([paths, build], S0),
-    % FIXME: Check actual state of checkout instead of just checking for folder
-    % presence!
-    case filelib:is_dir(filename:join(BuildPath, ".git")) of
-        false ->
+    case check_git_repository(BuildPath, S0) of
+        {false, S1}->
```

Comment:
It is a bit hard to follow what `true` and `false` means in these contexts. We should use dedicated atoms to convey meaning instead:

```erl
case check_git_repository(BuildPath, S0) of
    {up_to_date, S1} ->
        event(mapz:deep_put([build, download], false, S1), ['_skip']);
    {_Other, S1} ->
        ...
    end.
```